spec/models/like_spec.rb

```diff
@@ -0,0 +1,15 @@
+require 'rails_helper'
+
+RSpec.describe Like, type: :model do
+  describe 'associations' do
+    it 'belongs to post' do
+      association = described_class.reflect_on_association(:post)
+      expect(association.macro).to eq :belongs_to
+    end
+
+    it 'belongs to user' do
+      association = described_class.reflect_on_association(:user)
+      expect(association.macro).to eq :belongs_to
+    end
+  end
+end
```

Comment:
- [ ] [REQUIRED] The provided tests focus on the associations of the Like model but do not cover the custom method update_likes_counter and the validation. To fulfill the project instructions, you should add unit tests for the custom method and validation as stated by the project requirement.